src/policy/sft.rs

```diff
@@ -68,18 +65,7 @@ pub trait SFT {
     /// Some spaces, like `MallocSpace`, use third-party libraries to allocate memory.
     /// Such spaces needs to override this method.
     #[cfg(feature = "is_mmtk_object")]
-    #[inline(always)]
-    fn is_mmtk_object(&self, addr: Address) -> bool {
-        // Having found the SFT means the `addr` is in one of our spaces.
-        // Although the SFT map is allocated eagerly when the space is contiguous,
-        // the pages of the space itself are acquired on demand.
-        // Therefore, the page of `addr` may not have been mapped, yet.
-        if !addr.is_mapped() {
-            return false;
-        }
-        // The `addr` is mapped. We use the global alloc bit to get the exact answer.
-        alloc_bit::is_alloced_object(addr)
-    }
+    fn is_mmtk_object(&self, addr: Address) -> bool;
```

Comment:
Currently `SFTMap` is a global variable,
https://github.com/mmtk/mmtk-core/blob/644628bcc39e649e4125c092ad325bddbd24aaaf/src/mmtk.rs#L43
https://github.com/mmtk/mmtk-core/blob/644628bcc39e649e4125c092ad325bddbd24aaaf/src/policy/sft_map.rs#L271-L274

thus it cannot have a type parameter or an associated type. I don't know how hard it would be to refactor the global variable into an MMTk instance.

Reply:
The problem is that `SFT` needs to be object-safe.  It doesn't prevent it from having associated type, but every `dyn SFT` needs to be replaced with `SFT<VM = VM>`, which breaks the global SFTDenseChunkMap which is VM-agnostic.

The only solution I can think about is providing a global `fn is_mmtk_object<VM: VMBinding>(addr: Address) -> bool` which each Space implementation calls.  That's not better than calling `crate::util::alloc_bit::is_alloced_object` directly.  So I think it's probably OK to just leave it like this.